File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/oauth2/docs/openid/GlobalOpenIdAuthenticationMapper.kt
package io.micronaut.security.oauth2.docs.openid

//tag::clazz[]

import io.micronaut.context.annotation.Replaces
import io.micronaut.security.authentication.AuthenticationResponse
import io.micronaut.security.oauth2.endpoint.authorization.state.State
import io.micronaut.security.oauth2.endpoint.token.response.DefaultOpenIdAuthenticationMapper
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdAuthenticationMapper
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdClaims
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdTokenResponse
import jakarta.inject.Singleton
import org.reactivestreams.Publisher
import reactor.core.publisher.Flux

@Singleton
@Replaces(DefaultOpenIdAuthenticationMapper::class)
class GlobalOpenIdAuthenticationMapper : OpenIdAuthenticationMapper {

    override fun createAuthenticationResponse(providerName: String, tokenResponse: OpenIdTokenResponse, openIdClaims: OpenIdClaims, state: State?): Publisher<AuthenticationResponse> {
        return Flux.just(AuthenticationResponse.success("name"));
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/security/oauth2/docs/openid/GlobalOpenIdAuthenticationMapper.py
# tag::clazz[]
from jakarta.inject import Singleton
from micronaut.context.annotation import Replaces, Requires
from micronaut.security.authentication import AuthenticationResponse
from micronaut.security.oauth2.endpoint.authorization.state import State
from micronaut.security.oauth2.endpoint.token.response import DefaultOpenIdAuthenticationMapper, OpenIdAuthenticationMapper, OpenIdClaims, OpenIdTokenResponse
from org.reactivestreams import Publisher
from reactor.core.publisher import Flux


@Singleton
@Replaces(DefaultOpenIdAuthenticationMapper)
# end::clazz[]
@Requires(property="docs.classes")
# tag::clazz[]
class GlobalOpenIdAuthenticationMapper(OpenIdAuthenticationMapper):

    def createAuthenticationResponse(self, providerName: str, tokenResponse: OpenIdTokenResponse, openIdClaims: OpenIdClaims, state: State | None) -> Publisher[AuthenticationResponse]:
        return Flux.just(AuthenticationResponse.success("name"))
# end::clazz[]
